File: 2주차/두원사이의정수쌍/RightHennessy.py
# 1/4 을 구해보자
## a) 좌표를 모두 찍은 다음 원 사이에 존재하는지 확인
## b) 원 사이에 존재하는 점만 확인 - 이중 for문
## c) x좌표 찍고 y좌표 갯수 

import math

def solution(r1, r2):
    answer = countDots(r1, r2)
    return answer*4

def countDots(r1, r2):
    answer = 0
    for i in range(1, r2+1):
        answer += maximum(i, r2)+1 - minimum(i, r1)
    return answer

def minimum(i, r):
    value = r**2-i**2
    if value <= 0: return 0
    return int(math.ceil(math.sqrt(value)))

def maximum(i, r):
    return int(math.sqrt(r**2-i**2))

File: 10주차/k진수에서소수개수구하기/RightHennessy.kt
import kotlin.math.sqrt

// 타입 진짜 개팍친다..
class Solution {

    fun solution(n: Int, k: Int): Int {
        val convertedN = convertToK(n, k)
        val decimals = getDecimals(convertedN)

        return decimals.size
    }

    private fun convertToK(n: Int, k: Int): Long {
        if (k == 10) return n.toLong()

        var tmpN: Int = n
        var answer = ""
        while (tmpN >= k) {
            answer = (tmpN % k).toString() + answer
            tmpN /= k
        }
        if (tmpN != 0) {
            answer = tmpN.toString() + answer
        }
        return answer.toLong()
    }

    private fun getDecimals(number: Long): MutableList<Long> {
        val tmp = number.toString().split("0")
        val primes = mutableListOf<Long>()
        for (t in tmp) {
            runCatching {
                t.toLong()
            }.onSuccess {
                if (isPrime(it)) {
                    primes.add(it)
                }
            }
        }
        return primes
    }

    private fun isPrime(i: Long): Boolean {
        if (i <= 1) return false
        for (x: Long in 2.toLong()..sqrt(i.toDouble()).toLong()) {
            if (i.toLong() % x == 0L) return false
        }
        return true
    }
}